### PR TITLE
Maniacs Patch - Expand expression and control var ops support

### LINKED ISSUE #3654: Maniacs Patch: Math functions in a Variable Expression return incorrect values
Left is input value, middle is standalone ControlVariables/Math, right is ControlVariables/Expression.
Image

Demo Project:
[MathDemo.zip](https://github.com/user-attachments/files/30753326/MathDemo.zip)

### PR DESCRIPTION
**Pull request by jetrotal**
Adds Maniac Control Variables support for `Lerp`, `ArraySum`, `ArrayMin`, and `ArrayMax` in both interpreter command handling and expression function evaluation.

Fix #3654 

The expression parser now supports array/range/subscript values, multi-target inplace assignment behavior, and conditional execution paths (for logical/ternary operators).

Also remade some semantics rules to match maniacs: actor required EXP lookup, unsigned right-shift behavior, `Between` bounds logic, and array aggregation helpers in control variables.

Errors Parsing string variables are expect, they exist in the editor but not in maniacs player.

I quite didn't solved switch handling, that seems to be the last missing feature.

Also lifted expressions permissions in EasyRPG mode, as spoken with Ghabry.

Including a map and a huge TPC script with tests.

----------------------------------

Test Project: [Expressions_Test_Suite.zip](https://github.com/user-attachments/files/30682961/Expressions_Test_Suite.zip)

#### A list of prints detailing the tests:
Left: Maniacs Patch | Right EasyRPG.
Orange: Result Output | Yellow: expected result

<details>
<summary>View Tests Screenshots</summary>
image
image
image
image
image
image
image
👆 THOSE ARE THE LAST ONES THAT ARE NOT CORRECTLY IMPLEMENTED 👆
image
image
image
image
image
image
image
image
image
image
image
image
image
image
image
image

</details>

<details>
<summary> Test Event is easier to edit through this TPC script ( 900 LINES!!! ) </summary>

```python

// ============================================================================
// EXPRESSIONS USECASE TEST SUITE (100% COVERAGE)
// Covers all Maniac Patch VExpr features natively processed by the RM2k3 engine.
// Extensively refactored to exclude all compile-time meta operations and
// correctly test engine-side edge cases including raw bytecode fallbacks.
// ============================================================================

// ----------------------------------------------------------------------------
// Mode Selection (Prevents Battle-Only checks from crashing on Map)
// ----------------------------------------------------------------------------
defv is_battle_mode = 900
@msg.opt .opaq .bottom
@msg.show "Select Execution Mode:"
@msg.choice {
    .case "Map Mode (Safe)" {
        is_battle_mode = 0
    }
    .case "Battle Mode (Requires Battle)" {
        is_battle_mode = 1
    }
    .cancel {
        is_battle_mode = 0
    }
}

// ----------------------------------------------------------------------------
// Renderer Component
// ----------------------------------------------------------------------------
__fn renderPage $title {
    // Background 
    @pic[9].strpic {
        ""
        .pos 160, 120 .center
        .size 320, 240
        .rgbs 0, 0, 0, 0
        .chromakey 1 .skin "" .stretch
        .mapLayer 6 .eraseWhenTransfer
    }

    // Draw Title
    @pic[10].strpic {
        $title
        .pos 16, 4 .topLeft
        .font "TinyUnicode", 12 .spacing 0, 0 .noShadow
        .chromakey 1 .noframe .noPadding .nobg .noGradation
        .mapLayer 7 .eraseWhenTransfer
    }
    
    // Draw Col 1 (Expressions)
    @pic[11].strpic {
        t[11]
        .pos 16, 16 .topLeft
        .font "TinyUnicode", 12 .spacing 0, -1 .noShadow
        .chromakey 1 .noframe .noPadding .nobg .noGradation
        .mapLayer 7 .eraseWhenTransfer
    }
    
    // Get Col 1 Width and push X for Col 2
    @pic.getInfo [11] .currentRect(v[901], v[902], v[903], v[904]) .xywh
    v[905] = v[901] + v[903] + 8
    
    // Draw Col 2 (Results)
    v[999] = 16
    @pic[12].strpic {
        t[12]
        .pos v[905], v[999] .topLeft
        .font "TinyUnicode", 12 .spacing 0, -1 .noShadow
        .chromakey 1 .noframe .noPadding .nobg .noGradation
        .mapLayer 7 .eraseWhenTransfer
    }
    
    // Get Col 2 Width and push X for Col 3
    @pic.getInfo [12] .currentRect(v[901], v[902], v[903], v[904]) .xywh
    v[905] = v[901] + v[903] + 8
    
    // Draw Col 3 (Expected)
    v[999] = 16
    @pic[13].strpic {
        t[13]
        .pos v[905], v[999] .topLeft
        .font "TinyUnicode", 12 .spacing 0, -1 .noShadow
        .chromakey 1 .noframe .noPadding .nobg .noGradation
        .mapLayer 7 .eraseWhenTransfer
    }
    
    // Draw Footer
    @pic[14].strpic {
        "\C[1](Press Action to continue)\C[0]"
        .pos 260, 230 .center
        .font "TinyUnicode", 12 .spacing 0, -1 .noShadow
        .chromakey 1 .noframe .noPadding .nobg .noGradation
        .mapLayer 7 .eraseWhenTransfer
    }
    
    @wait .input
    @pic.erase [9..14]
}

// ----------------------------------------------------------------------------
// Page 1: Basic Arithmetic & Hex
// ----------------------------------------------------------------------------
v[1] = 10 + 5 * 2
v[2] = (10 + 5) * 2
v[3] = 250 / 10
v[4] = 14 % 5
v[5] = -10
v[6] = +10
v[7] = 0x0A + 0x05

t[11] .asg "10 + 5 * 2
(10 + 5) * 2
250 / 10
14 % 5
-10
+10
0x0A + 0x05"

t[12] .asg "=> \C[8]\v[1]\C[0]
=> \C[8]\v[2]\C[0]
=> \C[8]\v[3]\C[0]
=> \C[8]\v[4]\C[0]
=> \C[8]\v[5]\C[0]
=> \C[8]\v[6]\C[0]
=> \C[8]\v[7]\C[0]"

t[13] .asg "\C[4](20 - Precedence)\C[0]
\C[4](30)\C[0]
\C[4](25)\C[0]
\C[4](4)\C[0]
\C[4](-10)\C[0]
\C[4](10)\C[0]
\C[4](15 - Hex)\C[0]"

renderPage("\C[6]--- 1. Basic Arithmetic & Hex ---\C[0]")

// ----------------------------------------------------------------------------
// Page 2: Bitwise & Shifts
// ----------------------------------------------------------------------------
v[11] = 16 >> 2
v[12] = 3 << 2
v[13] = 5 | 2
v[14] = 5 & 3
v[15] = 5 ^ 3
v[16] = ~0

t[11] .asg "16 >> 2 (SHR)
3 << 2  (SHL)
5 | 2   (OR)
5 & 3   (AND)
5 ^ 3   (XOR)
~0      (NOT)"

t[12] .asg "=> \C[8]\v[11]\C[0]
=> \C[8]\v[12]\C[0]
=> \C[8]\v[13]\C[0]
=> \C[8]\v[14]\C[0]
=> \C[8]\v[15]\C[0]
=> \C[8]\v[16]\C[0]"

t[13] .asg "\C[4](4)\C[0]
\C[4](12)\C[0]
\C[4](7)\C[0]
\C[4](1)\C[0]
\C[4](6)\C[0]
\C[4](-1)\C[0]"

renderPage("\C[6]--- 2. Bitwise & Shifts ---\C[0]")

// ----------------------------------------------------------------------------
// Page 3: Logical & Relational
// ----------------------------------------------------------------------------
v[21] = (10 == 10)
v[22] = (10 != 5)
v[23] = (15 < 20)
v[24] = (20 <= 20)
v[25] = (100 > 10)
v[26] = (5 >= 1)
v[27] = (1 == 1) && (2 == 2)
v[28] = (1 == 0) || (2 == 2)
v[29] = !(1 == 1)

t[11] .asg "(10 == 10)
(10 != 5)
(15 < 20)
(20 <= 20)
(100 > 10)
(5 >= 1)
(1 == 1) && (2 == 2)
(1 == 0) || (2 == 2)
!(1 == 1)"

t[12] .asg "=> \C[8]\v[21]\C[0]
=> \C[8]\v[22]\C[0]
=> \C[8]\v[23]\C[0]
=> \C[8]\v[24]\C[0]
=> \C[8]\v[25]\C[0]
=> \C[8]\v[26]\C[0]
=> \C[8]\v[27]\C[0]
=> \C[8]\v[28]\C[0]
=> \C[8]\v[29]\C[0]"

t[13] .asg "\C[4](1)\C[0]
\C[4](1)\C[0]
\C[4](1)\C[0]
\C[4](1)\C[0]
\C[4](1)\C[0]
\C[4](1)\C[0]
\C[4](1)\C[0]
\C[4](1)\C[0]
\C[4](0)\C[0]"

renderPage("\C[6]--- 3. Logical & Relational ---\C[0]")

// ----------------------------------------------------------------------------
// Page 4: Unary Operations & String Parsing (VExpr)
// ----------------------------------------------------------------------------
t[41] .asg "404"
t[42] .asg "-50"
v[43] = 41
v[44] = t[41]       // tref -> parses "404" to 404
v[45] = t[v[43]]    // tdref -> parses "404" from t[41]
v[46] = t[42]       // parses "-50"
v[47] = 15
v[48] = -v[47]      // Unary Negative
v[49] = ~v[47]      // Bitwise NOT
v[50] = !v[47]      // Boolean NOT

t[11] .asg "v[44] = t[41]
v[45] = t[v[43]]
v[46] = t[42]
v[48] = -v[47]
v[49] = ~v[47]
v[50] = !v[47] (v[47]=15)"

t[12] .asg "=> \C[8]\v[44]\C[0]
=> \C[8]\v[45]\C[0]
=> \C[8]\v[46]\C[0]
=> \C[8]\v[48]\C[0]
=> \C[8]\v[49]\C[0]
=> \C[8]\v[50]\C[0]"

t[13] .asg "\C[4](404 - tref)\C[0]
\C[4](404 - tdref)\C[0]
\C[4](-50 - tref)\C[0]
\C[4](-15 - Unary Neg)\C[0]
\C[4](-16 - Bitwise NOT)\C[0]
\C[4](0 - Boolean NOT)\C[0]"

renderPage("\C[6]--- 4. Unary Operations & String Parsing ---\C[0]")

// ----------------------------------------------------------------------------
// Page 5: In-Place Assignments
// ----------------------------------------------------------------------------
v[51]=10
v[51]+=5
v[52]=10
v[52]-=5
v[53]=10
v[53]*=3
v[54]=30
v[54]/=3
v[55]=14
v[55]%=5
v[56]=3
v[56]<<=2
v[57]=16
v[57]>>=2
v[58]=5
v[58]|=2
v[59]=5
v[59]&=3
v[60]=5
v[60]^=3

t[11] .asg "v[51]=10 \C[5]+=\C[0] 5
v[52]=10 \C[5]-=\C[0] 5
v[53]=10 \C[5]*=\C[0] 3
v[54]=30 \C[5]/=\C[0] 3
v[55]=14 \C[5]%=\C[0] 5
v[56]=3  \C[5]<<=\C[0] 2
v[57]=16 \C[5]>>=\C[0] 2
v[58]=5  \C[5]|=\C[0] 2
v[59]=5  \C[5]&=\C[0] 3
v[60]=5  \C[5]^=\C[0] 3"

t[12] .asg "=> \C[8]\v[51]\C[0]
=> \C[8]\v[52]\C[0]
=> \C[8]\v[53]\C[0]
=> \C[8]\v[54]\C[0]
=> \C[8]\v[55]\C[0]
=> \C[8]\v[56]\C[0]
=> \C[8]\v[57]\C[0]
=> \C[8]\v[58]\C[0]
=> \C[8]\v[59]\C[0]
=> \C[8]\v[60]\C[0]"

t[13] .asg "\C[4](15)\C[0]
\C[4](5)\C[0]
\C[4](30)\C[0]
\C[4](10)\C[0]
\C[4](4)\C[0]
\C[4](12)\C[0]
\C[4](4)\C[0]
\C[4](7)\C[0]
\C[4](1)\C[0]
\C[4](6)\C[0]"

renderPage("\C[6]--- 5. In-Place Assignments ---\C[0]")

// ----------------------------------------------------------------------------
// Page 6: Advanced L-Values & Chaining
// ----------------------------------------------------------------------------
v[61] = 63
v[62] = 64
v[63] = 0
v[64] = 0
v[v[61]..v[62]] = 10  // Dynamic Range Scalar 
v[65] = 66
v[v[65]] = 20         // Indirect Var
v[v[65]] += 5         // Indirect Compound
v[67] = 68
s[v[67]] = 1          // Indirect Switch (Raw Fallback)
v[69] = v[70] = 30    // Chained Assignment
v[61] = 5 + (v[62] = 40) // Inline Assignment as Expr

t[11] .asg "v[v[61]..v[62]] = 10
v[v[65]] = 20; += 5
s[v[67]] = 1 (s[68])
v[69] = v[70] = 30
5 + (v[62] = 40)"

t[12] .asg "v[63]=\C[8]\v[63]\C[0], v[64]=\C[8]\v[64]\C[0]
v[66]=\C[8]\v[66]\C[0]
s[68]=\C[8]\s[68]\C[0]
v[69]=\C[8]\v[69]\C[0], v[70]=\C[8]\v[70]\C[0]
Ans=\C[8]\v[61]\C[0], v[62]=\C[8]\v[62]\C[0]"

t[13] .asg "\C[4](10, 10)\C[0]
\C[4](25 - Indirect)\C[0]
\C[4](1 - Ind. Switch)\C[0]
\C[4](30, 30 - Chained)\C[0]
\C[4](Ans=45, v[62]=40)\C[0]"

renderPage("\C[6]--- 6. Advanced L-Values & Chaining ---\C[0]")

// ----------------------------------------------------------------------------
// Page 7: Switch Ranges & Expressions
// ----------------------------------------------------------------------------
s[71..73].off
s[71..72].on          // Literal range switch batch
v[74] = 72
v[75] = 73
s[v[74]..v[75]].on    // Dynamic range switch batch (VExpr Raw Fallback)
v[76] = s[73]         // Reading switch as VExpr
v[77] = s[72].off     // Inline OFF evaluates to 0
v[78] = s[71].on      // Inline ON evaluates to 1
v[79] = s[73].toggle  // Dynamic Toggle evaluates to resulting state

t[11] .asg "s[71..72].on
s[v[74]..v[75]].on
v[76] = s[73]
v[77] = s[72].off
v[78] = s[71].on
v[79] = s[73].toggle"

t[12] .asg "s[71]=\C[8]\s[71]\C[0], s[72]=\C[8]\s[72]\C[0]
s[73]=\C[8]\s[73]\C[0]
=> \C[8]\v[76]\C[0]
=> \C[8]\v[77]\C[0] (s[72]=\C[8]\s[72]\C[0])
=> \C[8]\v[78]\C[0] (s[71]=\C[8]\s[71]\C[0])
=> \C[8]\v[79]\C[0] (s[73]=\C[8]\s[73]\C[0])"

t[13] .asg "\C[4]s[71..72] ON\C[0]
\C[4]s[73] ON (VExpr fallback)\C[0]
\C[4](1 - Read Switch)\C[0]
\C[4](0) (s[72] OFF)\C[0]
\C[4](1) (s[71] ON)\C[0]
\C[4](0) (s[73] OFF)\C[0]"

renderPage("\C[6]--- 7. Switch Ranges & Expressions ---\C[0]")

// ----------------------------------------------------------------------------
// Page 8: Built-in Math: Basic
// ----------------------------------------------------------------------------
v[81] = rnd(10, 20)
v[82] = min(16, 15)
v[83] = max(16, 15)
v[84] = clamp(16, 12, 18)
v[85] = abs(-50)
v[86] = between(15, 10, 20)
v[87] = lerp(100, 200, 1, 2)
v[88] = clamp(v[85], 0, 25)

t[11] .asg "rnd(10, 20)
min(16, 15)
max(16, 15)
clamp(16, 12, 18)
abs(-50)
between(15, 10, 20)
lerp(100, 200, 1, 2)
clamp(v[85], 0, 25)"

t[12] .asg "=> \C[8]\v[81]\C[0]
=> \C[8]\v[82]\C[0]
=> \C[8]\v[83]\C[0]
=> \C[8]\v[84]\C[0]
=> \C[8]\v[85]\C[0]
=> \C[8]\v[86]\C[0]
=> \C[8]\v[87]\C[0]
=> \C[8]\v[88]\C[0]"

t[13] .asg "\C[4](10~20)\C[0]
\C[4](15)\C[0]
\C[4](16)\C[0]
\C[4](16)\C[0]
\C[4](50)\C[0]
\C[4](1 - True)\C[0]
\C[4](150)\C[0]
\C[4](25 - With Variables)\C[0]"

renderPage("\C[6]--- 8. Built-in Math: Basic ---\C[0]")

// ----------------------------------------------------------------------------
// Page 9: Built-in Math: Advanced
// ----------------------------------------------------------------------------
v[91] = pow(2, 8)
v[92] = sqrt(100, 1)
v[93] = sin(90, 1, 100)
v[94] = cos(90, 1, 100)
v[95] = atan2(10, 10, 1)
v[96] = muldiv(100, 5, 2)
v[97] = divmul(100, 2, 5)

t[11] .asg "pow(2, 8)
sqrt(100, 1)
sin(90, 1, 100)
cos(90, 1, 100)
atan2(10, 10, 1)
muldiv(100, 5, 2)
divmul(100, 2, 5)"

t[12] .asg "=> \C[8]\v[91]\C[0]
=> \C[8]\v[92]\C[0]
=> \C[8]\v[93]\C[0]
=> \C[8]\v[94]\C[0]
=> \C[8]\v[95]\C[0]
=> \C[8]\v[96]\C[0]
=> \C[8]\v[97]\C[0]"

t[13] .asg "\C[4](256)\C[0]
\C[4](10)\C[0]
\C[4](100)\C[0]
\C[4](0)\C[0]
\C[4](45)\C[0]
\C[4](250)\C[0]
\C[4](250)\C[0]"

renderPage("\C[6]--- 9. Built-in Math: Advanced ---\C[0]")

// ----------------------------------------------------------------------------
// Page 10: Built-in Math: Aggregation
// ----------------------------------------------------------------------------
v[101..105] = [10, 20, 30, 40, 50]
v[106] = sum(101, 5)
v[107] = amin(101, 5)
v[108] = amax(101, 5)

t[11] .asg "v[101..105] = [10, 20, 30, 40, 50]
sum(101, 5)
amin(101, 5)
amax(101, 5)"

t[12] .asg "
=> \C[8]\v[106]\C[0]
=> \C[8]\v[107]\C[0]
=> \C[8]\v[108]\C[0]"

t[13] .asg "
\C[4](150)\C[0]
\C[4](10)\C[0]
\C[4](50)\C[0]"

renderPage("\C[6]--- 10. Built-in Math: Aggregation ---\C[0]")

// ----------------------------------------------------------------------------
// Page 11: Pointers & Indirect Refs
// ----------------------------------------------------------------------------
v[111] = 112
v[112] = 113
v[113] = 999
v[114] = v[v[111]]
v[115] = v[v[v[111]]]
v[116] = 117
s[117].on
v[118] = s[v[116]]

t[11] .asg "v[111]=112
v[112]=113
v[113]=999
v[114] = \C[5]v[v[111]]\C[0]
v[115] = \C[5]v[v[v[111]]]\C[0]
v[116]=117
s[117].on
v[118] = \C[5]s[v[116]]\C[0]"

t[12] .asg "


=> \C[8]\v[114]\C[0]
=> \C[8]\v[115]\C[0]


=> \C[8]\v[118]\C[0]"

t[13] .asg "


\C[4](113 - vdref)\C[0]
\C[4](999 - Deep Ptr)\C[0]


\C[4](1 - sdref)\C[0]"

renderPage("\C[6]--- 11. Pointers & Indirect Refs ---\C[0]")

// ----------------------------------------------------------------------------
// Page 12: Vectors & Array Literals Edge Cases
// ----------------------------------------------------------------------------
v[121] = [10, 20, 30][5]   // OOB Indexing -> 0
v[122] = [10, 20, 30][-1]  // Negative OOB -> 0
v[123..125] = [99, 88]     // Mismatch under -> 99, 88, 0
v[126..127] = [11, 22, 33] // Mismatch over -> 11, 22
v[128] = 2
v[129] = [77, 88, 99][v[128]] // Var Indexing

t[11] .asg "[10, 20, 30][5] (OOB)
[10, 20, 30][-1] (OOB)
v[123..125] = [99, 88] (Under)
v[126..127] = [11, 22, 33] (Over)
[77, 88, 99][v[128]] (v=2)"

t[12] .asg "=> \C[8]\v[121]\C[0]
=> \C[8]\v[122]\C[0]
=> \C[8]\v[123]\C[0], \C[8]\v[124]\C[0], \C[8]\v[125]\C[0]
=> \C[8]\v[126]\C[0], \C[8]\v[127]\C[0]
=> \C[8]\v[129]\C[0]"

t[13] .asg "\C[4](0 - Out of bounds)\C[0]
\C[4](0 - Out of bounds)\C[0]
\C[4](99, 88, 0)\C[0]
\C[4](11, 22)\C[0]
\C[4](99 - Var Indexing)\C[0]"

renderPage("\C[6]--- 12. Vectors & Array Literals Edge Cases ---\C[0]")

// ----------------------------------------------------------------------------
// Page 13: Ternary Operator
// ----------------------------------------------------------------------------
v[131] = 1 ? 10 : 20
v[132] = 0 ? 10 : 20
s[133].on
s[134].off
v[135] = s[133] ? s[134] ? 10 : 20 : 30
v[136] = s[134] ? 10 : s[133] ? 20 : 30
v[137] = s[133] ? (v[131] * 2) : 99 // Complex return

t[11] .asg "1 ? 10 : 20
0 ? 10 : 20
s[133]=ON, s[134]=OFF
s[133] ? s[134] ? 10 : 20 : 30
s[134] ? 10 : s[133] ? 20 : 30
s[133] ? (v[131] * 2) : 99"

t[12] .asg "=> \C[8]\v[131]\C[0]
=> \C[8]\v[132]\C[0]

=> \C[8]\v[135]\C[0]
=> \C[8]\v[136]\C[0]
=> \C[8]\v[137]\C[0]"

t[13] .asg "\C[4](10)\C[0]
\C[4](20)\C[0]

\C[4](20 - Nested Ternary)\C[0]
\C[4](20 - Nested Ternary)\C[0]
\C[4](20 - Expr Return)\C[0]"

renderPage("\C[6]--- 13. Ternary Operator ---\C[0]")

// ----------------------------------------------------------------------------
// Page 14: System Properties I
// ----------------------------------------------------------------------------
v[141] = sys.money
v[142] = sys.timer1
v[143] = sys.timer2
v[144] = sys.memberCount
v[145] = sys.saveCount
v[146] = sys.battleCount
v[147] = sys.winCount

t[11] .asg "sys.money
sys.timer1
sys.timer2
sys.memberCount
sys.saveCount
sys.battleCount
sys.winCount"

t[12] .asg "=> \C[8]\v[141]\C[0]
=> \C[8]\v[142]\C[0]
=> \C[8]\v[143]\C[0]
=> \C[8]\v[144]\C[0]
=> \C[8]\v[145]\C[0]
=> \C[8]\v[146]\C[0]
=> \C[8]\v[147]\C[0]"

t[13] .asg ""

renderPage("\C[6]--- 14. System Properties I ---\C[0]")

// ----------------------------------------------------------------------------
// Page 15: System Properties II
// ----------------------------------------------------------------------------
v[151] = sys.loseCount
v[152] = sys.escapeCount
v[153] = sys.tick
v[154] = sys.date
v[155] = sys.time
v[156] = sys.frame
v[157] = sys.version

t[11] .asg "sys.loseCount
sys.escapeCount
sys.tick
sys.date
sys.time
sys.frame
sys.version"

t[12] .asg "=> \C[8]\v[151]\C[0]
=> \C[8]\v[152]\C[0]
=> \C[8]\v[153]\C[0]
=> \C[8]\v[154]\C[0]
=> \C[8]\v[155]\C[0]
=> \C[8]\v[156]\C[0]
=> \C[8]\v[157]\C[0]"

t[13] .asg ""

renderPage("\C[6]--- 15. System Properties II ---\C[0]")

// ----------------------------------------------------------------------------
// Page 16: Map, Event & Item Properties
// ----------------------------------------------------------------------------
v[161] = ev[10001].mapId
v[162] = ev[10001].x
v[163] = ev[10001].y
v[164] = ev[10001].dir
v[165] = ev[10005].id
v[166] = item[1].count
v[167] = item[1].eqCount
v[168] = ev[10001].scrx
v[169] = ev[10001].scry

t[11] .asg "ev[10001].mapId (Player)
ev[10001].x
ev[10001].y
ev[10001].dir
ev[10005].id (Self)
item[1].count
item[1].eqCount
ev[10001].scrx
ev[10001].scry"

t[12] .asg "=> \C[8]\v[161]\C[0]
=> \C[8]\v[162]\C[0]
=> \C[8]\v[163]\C[0]
=> \C[8]\v[164]\C[0]
=> \C[8]\v[165]\C[0]
=> \C[8]\v[166]\C[0]
=> \C[8]\v[167]\C[0]
=> \C[8]\v[168]\C[0]
=> \C[8]\v[169]\C[0]"

t[13] .asg ""

renderPage("\C[6]--- 16. Map, Event & Item Properties ---\C[0]")

// ----------------------------------------------------------------------------
// Page 17: Vehicle Properties
// ----------------------------------------------------------------------------
v[171] = ev[10002].x
v[172] = ev[10003].y
v[173] = ev[10004].dir
v[174] = ev[10002].mapId
v[175] = ev[10003].scrx
v[176] = ev[10004].scry

t[11] .asg "ev[10002].x (Boat)
ev[10003].y (Ship)
ev[10004].dir (Airship)
ev[10002].mapId
ev[10003].scrx
ev[10004].scry"

t[12] .asg "=> \C[8]\v[171]\C[0]
=> \C[8]\v[172]\C[0]
=> \C[8]\v[173]\C[0]
=> \C[8]\v[174]\C[0]
=> \C[8]\v[175]\C[0]
=> \C[8]\v[176]\C[0]"

t[13] .asg ""

renderPage("\C[6]--- 17. Vehicle Properties ---\C[0]")

// ----------------------------------------------------------------------------
// Page 18: Actor Properties I
// ----------------------------------------------------------------------------
v[181] = actor[1].id
v[182] = actor[1].level
v[183] = actor[1].exp
v[184] = actor[1].reqExp
v[185] = actor[1].hp
v[186] = actor[1].mp
v[187] = actor[1].mhp
v[188] = actor[1].mmp
v[189] = actor[1].gauge

t[11] .asg "actor[1].id
actor[1].level
actor[1].exp
actor[1].reqExp
actor[1].hp
actor[1].mp
actor[1].mhp
actor[1].mmp
actor[1].gauge"

t[12] .asg "=> \C[8]\v[181]\C[0]
=> \C[8]\v[182]\C[0]
=> \C[8]\v[183]\C[0]
=> \C[8]\v[184]\C[0]
=> \C[8]\v[185]\C[0]
=> \C[8]\v[186]\C[0]
=> \C[8]\v[187]\C[0]
=> \C[8]\v[188]\C[0]
=> \C[8]\v[189]\C[0]"

t[13] .asg ""

renderPage("\C[6]--- 18. Actor Properties I ---\C[0]")

// ----------------------------------------------------------------------------
// Page 19: Actor Properties II
// ----------------------------------------------------------------------------
v[191] = actor[1].atk
v[192] = actor[1].def
v[193] = actor[1].mag
v[194] = actor[1].spd
v[195] = actor[1].weapon
v[196] = actor[1].shield
v[197] = actor[1].armor
v[198] = actor[1].helm
v[199] = actor[1].accessory

t[11] .asg "actor[1].atk
actor[1].def
actor[1].mag
actor[1].spd
actor[1].weapon
actor[1].shield
actor[1].armor
actor[1].helm
actor[1].accessory"

t[12] .asg "=> \C[8]\v[191]\C[0]
=> \C[8]\v[192]\C[0]
=> \C[8]\v[193]\C[0]
=> \C[8]\v[194]\C[0]
=> \C[8]\v[195]\C[0]
=> \C[8]\v[196]\C[0]
=> \C[8]\v[197]\C[0]
=> \C[8]\v[198]\C[0]
=> \C[8]\v[199]\C[0]"

t[13] .asg ""

renderPage("\C[6]--- 19. Actor Properties II ---\C[0]")

// ----------------------------------------------------------------------------
// Page 20: Party Member Properties I
// ----------------------------------------------------------------------------
v[201] = member[1].id
v[202] = member[1].level
v[203] = member[1].exp
v[204] = member[1].reqExp
v[205] = member[1].hp
v[206] = member[1].mp
v[207] = member[1].mhp
v[208] = member[1].mmp
v[209] = member[1].gauge

t[11] .asg "member[1].id
member[1].level
member[1].exp
member[1].reqExp
member[1].hp
member[1].mp
member[1].mhp
member[1].mmp
member[1].gauge"

t[12] .asg "=> \C[8]\v[201]\C[0]
=> \C[8]\v[202]\C[0]
=> \C[8]\v[203]\C[0]
=> \C[8]\v[204]\C[0]
=> \C[8]\v[205]\C[0]
=> \C[8]\v[206]\C[0]
=> \C[8]\v[207]\C[0]
=> \C[8]\v[208]\C[0]
=> \C[8]\v[209]\C[0]"

t[13] .asg ""

renderPage("\C[6]--- 20. Party Member Properties I ---\C[0]")

// ----------------------------------------------------------------------------
// Page 21: Party Member Properties II
// ----------------------------------------------------------------------------
v[211] = member[1].atk
v[212] = member[1].def
v[213] = member[1].mag
v[214] = member[1].spd
v[215] = member[1].weapon
v[216] = member[1].shield
v[217] = member[1].armor
v[218] = member[1].helm
v[219] = member[1].accessory

t[11] .asg "member[1].atk
member[1].def
member[1].mag
member[1].spd
member[1].weapon
member[1].shield
member[1].armor
member[1].helm
member[1].accessory"

t[12] .asg "=> \C[8]\v[211]\C[0]
=> \C[8]\v[212]\C[0]
=> \C[8]\v[213]\C[0]
=> \C[8]\v[214]\C[0]
=> \C[8]\v[215]\C[0]
=> \C[8]\v[216]\C[0]
=> \C[8]\v[217]\C[0]
=> \C[8]\v[218]\C[0]
=> \C[8]\v[219]\C[0]"

t[13] .asg ""

renderPage("\C[6]--- 21. Party Member Properties II ---\C[0]")

// ----------------------------------------------------------------------------
// Page 22: Enemy Properties (Battle Only)
// ----------------------------------------------------------------------------
@if `is_battle_mode == 1 bl {
    v[221] = enemy[1].id
    v[222] = enemy[1].hp
    v[223] = enemy[1].mp
    v[224] = enemy[1].mhp
    v[225] = enemy[1].mmp
    v[226] = enemy[1].atk
    v[227] = enemy[1].def
    v[228] = enemy[1].mag
    v[229] = enemy[1].spd
    v[230] = enemy[1].gauge

    t[11] .asg "enemy[1].id
enemy[1].hp
enemy[1].mp
enemy[1].mhp
enemy[1].mmp
enemy[1].atk
enemy[1].def
enemy[1].mag
enemy[1].spd
enemy[1].gauge"

    t[12] .asg "=> \C[8]\v[221]\C[0]
=> \C[8]\v[222]\C[0]
=> \C[8]\v[223]\C[0]
=> \C[8]\v[224]\C[0]
=> \C[8]\v[225]\C[0]
=> \C[8]\v[226]\C[0]
=> \C[8]\v[227]\C[0]
=> \C[8]\v[228]\C[0]
=> \C[8]\v[229]\C[0]
=> \C[8]\v[230]\C[0]"

    t[13] .asg ""
    renderPage("\C[6]--- 22. Enemy Properties (Battle) ---\C[0]")
} .else bl {
    t[11] .asg "Enemy properties skipped."
    t[12] .asg ""
    t[13] .asg ""
    renderPage("\C[6]--- 22. Enemy Properties (Battle) ---\C[0]")
}

// ----------------------------------------------------------------------------
// Page 23: Edge Cases (Short-Circuit & Zero-Div)
// ----------------------------------------------------------------------------
v[231] = 0
v[232] = ((1 == 0) && (v[231] = 5)) ? 1 : 0
v[233] = 0
v[234] = ((1 == 1) || (v[233] = 5)) ? 1 : 0

v[235] = -10 / 3
v[236] = -10 % 3

v[237] = 0
v[238] = 100 / v[237]
v[239] = 100 % v[237]

t[11] .asg "((1 == 0) && (v[231] = 5)) ? 1 : 0
Did v[231] mutate?
((1 == 1) || (v[233] = 5)) ? 1 : 0
Did v[233] mutate?
-10 / 3
-10 % 3
100 / v[237]  (v=0)
100 % v[237]  (v=0)"

t[12] .asg "=> \C[8]\v[232]\C[0]
=> \C[8]\v[231]\C[0]
=> \C[8]\v[234]\C[0]
=> \C[8]\v[233]\C[0]
=> \C[8]\v[235]\C[0]
=> \C[8]\v[236]\C[0]
=> \C[8]\v[238]\C[0]
=> \C[8]\v[239]\C[0]"

t[13] .asg "\C[4](0)\C[0]
\C[4](0 - Short-Circuit)\C[0]
\C[4](1)\C[0]
\C[4](0 - Short-Circuit)\C[0]
\C[4](-3)\C[0]
\C[4](-1)\C[0]
\C[4](100 - Zero Guard)\C[0]
\C[4](0 - Zero Guard)\C[0]"

renderPage("\C[6]--- 23. Edge Cases (Short-Circuit & Zero-Div) ---\C[0]")
```
</details>

### CODE REVIEW

File: src/game_interpreter_control_variables.cpp [hunk 1]
```diff
@@ -421,11 +429,11 @@ int ControlVariables::Binary(int op, int arg1, int arg2) {
 				result = arg1_64;
 			}
 			break;
-		case 5:
+		case 5:result = arg1_64 >> arg2_64;
```
Comment (Ghabry) [inline src/game_interpreter_control_variables.cpp]: Copy paste error? (Delete the line)

**Comment by Ghabry** [comment]
The Underflow test was also failing. Fixing this also fixed the switch test for me :).

In some parts the code style somehow completely messed up. Fixed all of it and looks good to me now.

**Review by Ghabry — APPROVED** [review]

**Comment by Ghabry** [comment]
There is another feature we don't support but this looks slightly more involved to solve due to how our code is structured:

```
v[100..110] = 42
```

This returns a range to the interpreter. We currently only use the first value and ignore the rest.
